### DISCUSSION TITLE
Way to prioritize keybindings from .ideavimrc

### DISCUSSION THREAD

**Discussion by wolfisdev** [JetBrains/ideavim#302] [Q&A] [answered]
I love IdeaVim, been using it for years and never knew you could have a .ideavimrc config. Upon finding out hours ago, I wrote up a config file with a ton of keybindings.

Question: Is there a way to have the JetBrains IDEs prioritize the .ideavimrc keybindings over the ones set in the keymap?
Because usually when a .ideavimrc keybind did not work, it was because the keybinding was configured in the IDE keymap. So, unbinding it from the keymap makes the .ideavimrc keybinding work. But since I use different IDEs (IntelliJ, PyCharm, WebStorm, etc.) and different computers, it would be absolutely amazing to just have this one .ideavimrc file that is read by all IDEs and not to have to go to each IDE and manually delete each conflicting keybinding from the keymap.

**Comment by AlexPl292** [comment, marked as answer]
Hi! You can configure handlers for shortcuts in `Settings | Editor | Vim Emulation`. In addition, right now I'm working on support for per-mode shortcuts handlers (e.g. use IDE shortcut in normal mode and vim shortcut in insert mode) and support for defining all handlers in `~/.ideavimrc` file. Just wait a little bit, we'll bring you the new release with this feature in one-two weeks :)

**Reply by wolfisdev** [reply]
Cool, I knew about the `Vim Emulation` settings, but that was exactly what I was trying to avoid, having to decide for each individual mapping. Also, I am not quite sure If `Vim Emulation` settings automatically pick up on all mapping collisions (haven't tested it thoroughly). I mean whether it picks up collisions with custom mappings from the `~/.ideavimrc` as well or just the collisions of the default vim mappigns.

But being able to define the handlers in the `~/.ideavimrc` is probably what I am looking for, especially if it'll work for custom mappings in the rc file. Even better if the handler can be defined together with the mapping, so you don't have to maintain a separate mapping and handler list ;).

Looking forward to it, Thanks!
